File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/participant-assignment-track-quiz-question-answer.model.ts
import { AssignmentQuestionAnswerValue } from './quiz-assignment-form-question.model';

export interface IParticipantAssignmentTrackQuizQuestionAnswer {
  id: string;
  quizAssignmentFormQuestionId: string;
  quizAnswerId: string;
  answerValue: AssignmentQuestionAnswerValue;
  manualScore?: number;
  manualScoredBy?: string;
  score?: number;
  scoreBy?: string;
  submittedDate?: Date;
}

export class ParticipantAssignmentTrackQuizQuestionAnswer implements IParticipantAssignmentTrackQuizQuestionAnswer {
  public id: string;
  public quizAssignmentFormQuestionId: string;
  public quizAnswerId: string;
  public answerValue: AssignmentQuestionAnswerValue;
  public manualScore?: number;
  public manualScoredBy?: string;
  public score?: number;
  public scoreBy?: string;
  public submittedDate?: Date;
  constructor(data?: IParticipantAssignmentTrackQuizQuestionAnswer) {
    if (data) {
      this.id = data.id;
      this.quizAssignmentFormQuestionId = data.quizAssignmentFormQuestionId;
      this.quizAnswerId = data.quizAnswerId;
      this.answerValue = data.answerValue;
      this.manualScore = data.manualScore;
      this.manualScoredBy = data.manualScoredBy;
      this.score = data.score;
      this.scoreBy = data.scoreBy;
      this.submittedDate = data.submittedDate ? new Date(data.submittedDate) : undefined;
    }
  }

  public get givedScore(): number {
    return this.manualScore != null ? this.manualScore : this.score;
  }
}

File: pdlearning/src/microservices/course-microservice/Api/Application/Models/ParticipantAssignmentTrackQuizQuestionAnswerModel.cs
using System;
using Microservice.Course.Domain.Entities;

namespace Microservice.Course.Application.Models
{
    public class ParticipantAssignmentTrackQuizQuestionAnswerModel
    {
        public ParticipantAssignmentTrackQuizQuestionAnswerModel(ParticipantAssignmentTrackQuizQuestionAnswer questionAnswer)
        {
            Id = questionAnswer.Id;
            QuizAssignmentFormQuestionId = questionAnswer.QuizAssignmentFormQuestionId;
            QuizAnswerId = questionAnswer.QuizAnswerId;
            AnswerValue = questionAnswer.AnswerValue;
            ManualScore = questionAnswer.ManualScore;
            ManualScoredBy = questionAnswer.ManualScoredBy;
            Score = questionAnswer.Score;
            SubmittedDate = questionAnswer.SubmittedDate;
        }

        public int Id { get; set; }

        public Guid QuizAssignmentFormQuestionId { get; set; }

        public Guid QuizAnswerId { get; set; }

        public object AnswerValue { get; set; }

        public float? ManualScore { get; set; }

        public Guid? ManualScoredBy { get; set; }

        public float? Score { get; set; }

        public DateTime? SubmittedDate { get; set; }
    }
}
